### PR TITLE
Colour support verification

### PR DESCRIPTION
**Pull request by eamansour**
For https://github.com/galasa-dev/projectmanagement/issues/1248

Changes:
- Added IVTs to test colour support against CBSA
- Additional unit tests
- Added an initial version of the modify field order to resolve an issue encountered by the CiP team where the order was not being recognised during data stream processing

### CODE REVIEW

**Comment by techcobweb** [comment]
Where are we on this story please ? Will it close today ? Do you need any reviewers assigned ?

**Comment by eamansour** [comment]
> Where are we on this story please ? Will it close today ? Do you need any reviewers assigned ?

These are the main changes needed for the story, I'll be opening a PR for some minor 3270 manager documentation updates soon and that should close the story.

**Comment by galasa-team** [comment]
Build successful

**Review by techcobweb — APPROVED** [review]
Looks great to me. Like the new unit tests.

**Review by hobbit1983 — CHANGES_REQUESTED** [review]
It's all good, just small change to the IVT

File: galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos3270.manager.ivt/src/main/java/dev/galasa/zos3270/manager/ivt/Zos3270IVT.java [hunk 1]
```diff
@@ -167,8 +173,44 @@ public void driveWaitForTextInField() throws TextNotFoundException, ErrorTextFou
     		logger.info("No text found exception correctly thrown");
     		assertThat(tnfe.getMessage()).contains("Unable to find a field containing any of the request text");
     	}
-    	
-    	
     }
-    
+
+	@Test
+	public void cursorColourTest() throws Zos3270Exception, CoreManagerException {
```
[the remark this answers is not in the document]
Comment (eamansour) [reply galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos3270.manager.ivt/src/main/java/dev/galasa/zos3270/manager/ivt/Zos3270IVT.java]: Thanks - This is addressed in https://github.com/galasa-dev/managers/pull/818/commits/c988f3011938328854738b6d1688052ffdcf72f7

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

File: galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos3270.manager.ivt/src/main/java/dev/galasa/zos3270/manager/ivt/Zos3270IVT.java [hunk 2]
```diff
@@ -46,8 +48,12 @@ public class Zos3270IVT {
     public String providedRunName;
     private String runName  = new String();
     
+	private String credentialsId = "PRIMARY";
     private String applid = "IYK2ZNB5";
 
+	@TestProperty(prefix = "IVT.REGION", suffix = "APPLID")
+	public String cbsaApplid;
+
```
Comment (hobbit1983) [inline galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos3270.manager.ivt/src/main/java/dev/galasa/zos3270/manager/ivt/Zos3270IVT.java:56]: add required attribute - set to false

File: galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos3270.manager.ivt/src/main/java/dev/galasa/zos3270/manager/ivt/Zos3270IVT.java [hunk 3]
```diff
@@ -46,8 +48,12 @@ public class Zos3270IVT {
...
     @Test
     public void checkInjection() {
         assertThat(logger).as("Logger Field").isNotNull();
```
Comment (hobbit1983) [inline galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos3270.manager.ivt/src/main/java/dev/galasa/zos3270/manager/ivt/Zos3270IVT.java:59]: can we remove the isNotNull assertions

File: galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos3270.manager.ivt/src/main/java/dev/galasa/zos3270/manager/ivt/Zos3270IVT.java [hunk 4]
```diff
@@ -46,8 +48,12 @@ public class Zos3270IVT {
     public String providedRunName;
```
Comment (hobbit1983) [inline galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos3270.manager.ivt/src/main/java/dev/galasa/zos3270/manager/ivt/Zos3270IVT.java:48]: delete providedRunName

**Review by hobbit1983 — CHANGES_REQUESTED** [review]

File: galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos3270.manager.ivt/src/main/java/dev/galasa/zos3270/manager/ivt/Zos3270IVT.java (hunk 2 above)
Comment (eamansour) [reply galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos3270.manager.ivt/src/main/java/dev/galasa/zos3270/manager/ivt/Zos3270IVT.java]: Resolved in https://github.com/galasa-dev/managers/pull/818/commits/31ddf99e10cc7d6e738c9d00daccdf572ad412c0

File: galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos3270.manager.ivt/src/main/java/dev/galasa/zos3270/manager/ivt/Zos3270IVT.java (hunk 3 above)
Comment (eamansour) [reply galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos3270.manager.ivt/src/main/java/dev/galasa/zos3270/manager/ivt/Zos3270IVT.java]: Resolved in https://github.com/galasa-dev/managers/pull/818/commits/31ddf99e10cc7d6e738c9d00daccdf572ad412c0

File: galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos3270.manager.ivt/src/main/java/dev/galasa/zos3270/manager/ivt/Zos3270IVT.java (hunk 4 above)
Comment (eamansour) [reply galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos3270.manager.ivt/src/main/java/dev/galasa/zos3270/manager/ivt/Zos3270IVT.java]: Resolved in https://github.com/galasa-dev/managers/pull/818/commits/31ddf99e10cc7d6e738c9d00daccdf572ad412c0

**Comment by galasa-team** [comment]
Build successful

**Review by hobbit1983 — APPROVED** [review]

**Pull request merged**